circle_center/main.ipynb

```diff
@@ -0,0 +1,115 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "6616ba09-f224-4f44-a869-9d4bd1e05faa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import re"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "e8ba2346-a991-4d2c-9c41-1fe61c16b1f2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def find_centers(input_geometries):\n",
+    "    \"\"\"\n",
+    "    Process a list of CIRCULARSTRING geometries and compute the center of each circular arc.\n",
+    "\n",
+    "    Parameters:\n",
+    "    - input_geometries (list): A list of strings containing CIRCULARSTRING geometries.\n",
+    "\n",
+    "    Returns:\n",
+    "    - list: A list of dictionaries containing the computed center coordinates.\n",
+    "    \"\"\"\n",
+    "    centers = []  # Store centers for each geometry\n",
+    "\n",
+    "    for input_geometry in input_geometries:\n",
+    "        if not input_geometry.startswith(\"CIRCULARSTRING\"):\n",
+    "            raise ValueError(f'Invalid geometry: \"{input_geometry}\" must start with \"CIRCULARSTRING\".')\n",
+    "\n",
+    "        # Extract coordinates from the input\n",
+    "        coordinate_string = re.sub(r\"[()]\", \"\", input_geometry.replace(\"CIRCULARSTRING\", \"\").strip())\n",
```

Comment:
I would consider extracting the numbers with a regex - it might be a bit simpler and allow you to skip removing the "CIRCULARSTRING" text:
```python
numbers = [float(num) for num in re.findall(r'\d+\.?\d*', input_geometry)]
```